### ISSUE TITLE
USER STORY: Streamline task tab update code

### ISSUE THREAD

**Issue by blahosyl** [blahosyl/task-manager-frontend#132]
As a **Developer**,

I want to **avoid superfluous code in the task tab update functionality**,

so that **the application does not perform unnecessary actions**

### Acceptance Criteria

1. Task tab update code does not contain unnecessary parts.

### Tasks

- [x] check which props each tab needs (probably the only tab that requires all is the "watched by me" tab, as other aspects of the tasks cannot be changed on this page.

**Comment by blahosyl** [comment]
Related to #124

**Comment by blahosyl** [comment]
After testing the removal of the `changedWatch` prop from all tabs except for "watched by me", I realised that this causes a bug where the watched status of a task is only updated on the tab the user is currently on and the "watched" tab, but not on other tabs. Consequently, clicking the Watch icon on other tabs causes an error (since the user is now trying to watch an already watched task or unwatch an unwatched task).

Therefore, I rerolled removal of the `changedWatch` prop.
